tests/test_modeling_tf_common.py

```diff
@@ -990,15 +991,16 @@ def recursive_check(tuple_object, dict_object):
                 dict_inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
                 check_equivalence(model, tuple_inputs, dict_inputs, {"output_hidden_states": True})
 
-                tuple_inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
-                dict_inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
-                check_equivalence(model, tuple_inputs, dict_inputs, {"output_attentions": True})
+                if self.has_attentions:
```

Comment:
What happens for this test before this PR? It failed?

Reply:
Pretty much - a tensorflow model would fail when equivalent tests to a PyTorch model without attention layers were implemented.  The same logic I'm adding here are in `test_modeling_common.py`. I found this when working on the ResNet port. All of the models this affects only have PyTorch implementations[ apart from convnext which uses unittest.skip](https://github.com/huggingface/transformers/blob/b1160c0b565d7f03342cfd14b780d7bbd802401e/tests/models/convnext/test_modeling_tf_convnext.py#L173) in its tests, so there's no failing tests at the moment on main.